pkg/server/plugin/datastore/sql/mysql.go

```diff
@@ -1,25 +1,96 @@
 package sql
 
 import (
+	"crypto/tls"
+	"crypto/x509"
 	"errors"
+	"io/ioutil"
 
 	mysqldriver "github.com/go-sql-driver/mysql"
 	"github.com/jinzhu/gorm"
+	"github.com/sirupsen/logrus"
 
 	// gorm mysql dialect init registration
 	_ "github.com/jinzhu/gorm/dialects/mysql"
 )
 
 type mysql struct{}
 
-func (my mysql) connect(connectionString string) (*gorm.DB, error) {
-	db, err := gorm.Open("mysql", connectionString)
+const (
+	tlsConfigName = "spireCustomTLS"
+)
+
+func (my mysql) connect(cfg *configuration) (*gorm.DB, error) {
+	connString, err := configureConnection(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	db, err := gorm.Open("mysql", connString)
 	if err != nil {
 		return nil, err
 	}
 	return db, nil
 }
 
+// configureConnection modifies the connection string to support features that
+// normally require code changes, like custom Root CAs or client certificates
+func configureConnection(cfg *configuration) (string, error) {
+	if !hasTLSConfig(cfg) {
+		// connection string doesn't have to be modified
+		return cfg.ConnectionString, nil
+	}
+
+	tlsConf := tls.Config{}
+
+	opts, err := mysqldriver.ParseDSN(cfg.ConnectionString)
+	if err != nil {
+		// the connection string should have already been validated by now
+		// (in validateMySQLConfig)
+		return "", sqlError.Wrap(err)
+	}
+
+	// load and configure Root CA if it exists
+	if len(cfg.RootCAPath) > 0 {
+		logrus.Infof("configuring MySQL connection to use Root CA found in %s", cfg.RootCAPath)
```

Comment:
Couple thoughts on this (and subsequent) log line

First is since we are not using a logger passed in from core, if we're configured to log to file or if we are configured for level above info, this won't have any knowledge.  Second, @azdagron is currently working on plugin catalog refactor that will expose logging to plugins as a first class citizen, solving problem 1. Finally, since these messages aren't particularly critical, I might leave them out for now and we can add them later when we have fixed the plugin logging situation (but perhaps as debug?).

What do you think? If you feel strongly about these lines, then I recommend we ensure that info level is suitable, and then pass a logger into datastore plugin... but I am not sure it's worth it at this point. Curious to hear @azdagron thoughts too

Reply:
I think that makes sense. I actually added those log lines as an after-thought and mainly to help with debugging. I'm happy to take them out and revisit once we have loggers available to plugins. 👍